pkg/installer/filebuilder.go

```diff
@@ -0,0 +1,237 @@
+package installer
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/troubleshoot/cmd/util"
+	apiv1 "github.com/storageos/kubectl-storageos/api/v1"
+	pluginutils "github.com/storageos/kubectl-storageos/pkg/utils"
+	pluginversion "github.com/storageos/kubectl-storageos/pkg/version"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/kustomize/api/filesys"
+)
+
+// fileBuilder is used to hold data required to build a file in the in-memory fs
+type fileBuilder struct {
+	// yamlPath is passed via plugin flag, it may be a local
+	// file path, a URL or a docker repo URL.
+	yamlPath string
+	// yamlURL github release URL to yaml file
+	yamlUrl string
+	// yamlImage is a manifests image storing yaml file
+	yamlImage string
+	// fileName of yaml file
+	fileName string
+	// namespace of yaml file
+	namespace string
+}
+
+func newFileBuilder(yamlPath, yamlUrl, yamlImage, fileName, namespace string) *fileBuilder {
+	return &fileBuilder{
+		yamlPath:  yamlPath,
+		yamlUrl:   yamlUrl,
+		yamlImage: yamlImage,
+		fileName:  fileName,
+		namespace: namespace,
+	}
+}
+
+// buildInstallerFileSys builds an in-memory filesystem for installer with relevant storageos and
```

Comment:
Have you changed those lines above, or just moved it here?

Reply:
`fileBuilder` struct and `newFileBuilder()` are new, to save passing all these parameters around functions.